crates/toml_datetime/src/datetime.rs

```diff
@@ -513,6 +533,40 @@ impl<'de> de::Deserialize<'de> for Datetime {
     }
 }
 
+#[cfg(feature = "serde")]
+impl<'de> de::Deserialize<'de> for Date {
+    fn deserialize<D>(deserializer: D) -> Result<Date, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        match Datetime::deserialize(deserializer)? {
+            Datetime {
+                date: Some(date),
+                time: None,
+                offset: None,
+            } => Ok(date),
+            _ => Err(de::Error::custom("expected only date portion of datetime")),
```

Comment:
I feel like [`Error::invalid_type`](https://docs.rs/serde/latest/serde/de/trait.Error.html#method.invalid_type) would be more appropriate for these